corehq/apps/app_manager/util.py

```diff
@@ -595,3 +597,62 @@ def get_and_assert_practice_user_in_domain(practice_user_id, domain):
                 username=user.username)
         )
     return user
+
+
+class LatestAppInfo(object):
+
+    def __init__(self, brief_app_id, domain):
+        """
+        Wrapper to get latest app version and CommCare APK version info
+
+        args:
+            brief_app_id: id of an app that is not copy (to facilitate quickcaching)
+
+        raises Http404 error if id is not valid
+        raises assertion error if an id of app copy is passed
+        """
+        self.app_id = brief_app_id
+        self.domain = domain
+
+    @property
+    @memoized
+    def app(self):
+        app = get_app(self.domain, self.app_id)
+        # quickache based on a copy app_id will have to be updated too fast
+        assert not app.copy_of, "this class doesn't handle copy app ids"
```

Comment:
Generally, since this info will rarely change, I designed the views/utils in such a way that the info is cached based on the brief app id (provided by mobile) and pulling the app is avoided as much possible